lib/crypto-tools/keyring.js

```diff
@@ -273,4 +281,95 @@ KeyRing.prototype._decrypt = function(data) {
   return JSON.parse(dec) && dec;
 };
 
+/**
+ * Retrieve the bucketKey
+ * @param {String} bucketId - bucket id
+ * @returns {String|null}
+ */
+KeyRing.prototype.generateBucketKey = function(bucketId) {
+  if(!this._mnemonic || !bucketId){
+    return null;
+  }
+  return DeterministicKeyIv.getDeterministicKey(this._mnemonic, bucketId);
```

Comment:
Cool. Should we use [`mnemonic.toSeed()`](https://github.com/bitpay/bitcore-mnemonic/blob/master/lib/mnemonic.js#L178) here?